area and normalmap plugins don't compile with scalar_mono_polarized variant

## System configuration

Platform: macOS 10.15.5
Compiler: Apple clang version 11.0.3 (clang-1103.0.32.59)
Python version: Python 3.7.7 :: Anaconda, Inc.
Mitsuba 2 version: 695eefed46e9536a65ee61e7fc4f4134c303c682
Compiled variants:
* `scalar_mono_polarized`
* `scalar_rgb`

## Problem

Compilation of the `aera` and `normalmap` plugins fail when using the aforementioned variant selection. Here is the compiler's error message:
```
[206/420] Building CXX object src/bsdfs/CMakeFiles/normalmap-obj.dir/normalmap.cpp.o
FAILED: src/bsdfs/CMakeFiles/normalmap-obj.dir/normalmap.cpp.o
/usr/bin/clang++  -DLITTLE_ENDIAN -DMTS_ENABLE_PROFILER -D__OSX__ -I../include -I../ext/tinyformat -I../ext/enoki/include -I../ext/tbb/include -I../ext/pybind11/include -I/Users/leroyv/miniconda3/envs/mitsuba2/include/python3.7m -flto=thin  -stdlib=libc++ -D_LIBCPP_VERSION -std=gnu++17 -fcolor-diagnostics -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -fPIC   -fno-stack-protector -fomit-frame-pointer -fno-math-errno -ffp-contract=fast -march=native -Wall -Wextra -Wno-unused-local-typedefs -fvisibility=hidden -Wdouble-promotion -MD -MT src/bsdfs/CMakeFiles/normalmap-obj.dir/normalmap.cpp.o -MF src/bsdfs/CMakeFiles/normalmap-obj.dir/normalmap.cpp.o.d -o src/bsdfs/CMakeFiles/normalmap-obj.dir/normalmap.cpp.o -c ../src/bsdfs/normalmap.cpp
../src/bsdfs/normalmap.cpp:105:16: error: invalid operands to binary expression ('mitsuba::NormalMap<float, enoki::Matrix<mitsuba::Color<float, 1>, 4> >::Mask' (aka 'bool') and 'enoki::Mask<mitsuba::Color<float, 1>, 4>')
        active &= any(neq(weight, 0.f));
        ~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~
../src/bsdfs/normalmap.cpp:176:19: note: in instantiation of member function 'mitsuba::NormalMap<float, enoki::Matrix<mitsuba::Color<float, 1>, 4> >::sample' requested here
MTS_EXPORT_PLUGIN(NormalMap, "Normal map material adapter");
                  ^
../ext/enoki/include/enoki/array_router.h:203:1: note: candidate template ignored: requirement 'false' was not satisfied [with T1 = bool]
ENOKI_ROUTE_COMPOUND_OPERATOR(&)
^
../ext/enoki/include/enoki/array_router.h:155:22: note: expanded from macro 'ENOKI_ROUTE_COMPOUND_OPERATOR'
    ENOKI_INLINE T1 &operator op##=(T1 &a1, const T2 &a2) {                    \
                     ^
1 error generated.
[240/420] Building CXX object src/emitters/CMakeFiles/area-obj.dir/area.cpp.o
FAILED: src/emitters/CMakeFiles/area-obj.dir/area.cpp.o
/usr/bin/clang++  -DLITTLE_ENDIAN -DMTS_ENABLE_PROFILER -D__OSX__ -I../include -I../ext/tinyformat -I../ext/enoki/include -I../ext/tbb/include -I../ext/pybind11/include -I/Users/leroyv/miniconda3/envs/mitsuba2/include/python3.7m -flto=thin  -stdlib=libc++ -D_LIBCPP_VERSION -std=gnu++17 -fcolor-diagnostics -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -fPIC   -fno-stack-protector -fomit-frame-pointer -fno-math-errno -ffp-contract=fast -march=native -Wall -Wextra -Wno-unused-local-typedefs -fvisibility=hidden -Wdouble-promotion -MD -MT src/emitters/CMakeFiles/area-obj.dir/area.cpp.o -MF src/emitters/CMakeFiles/area-obj.dir/area.cpp.o.d -o src/emitters/CMakeFiles/area-obj.dir/area.cpp.o -c ../src/emitters/area.cpp
../src/emitters/area.cpp:99:61: error: no member named 'sample' in 'mitsuba::Texture<float, Spectrum<float, 4> >'
            std::tie(wavelength, spec_weight) = m_radiance->sample(
                                                ~~~~~~~~~~  ^
../src/emitters/area.cpp:168:19: note: in instantiation of member function 'mitsuba::AreaLight<float, mitsuba::Spectrum<float, 4> >::sample_ray' requested here
MTS_EXPORT_PLUGIN(AreaLight, "Point emitter")
                  ^
1 error generated.
```

## Steps to reproduce

1. Select the `scalar_mono_polarized` and `scalar_rgb` variants in `mitsuba.conf`.
2. Compile Mitsuba.

Now I just realise that I actually also had the `scalar_spectral` variant active. If I understand the compilation logs correctly, it's for this variant that `area` fails to compile. With only `scalar_rgb` and `scalar_spectral`, I get this:
```
[159/377] Building CXX object src/emitters/CMakeFiles/area-obj.dir/area.cpp.o
FAILED: src/emitters/CMakeFiles/area-obj.dir/area.cpp.o
/usr/bin/clang++  -DLITTLE_ENDIAN -DMTS_ENABLE_PROFILER -D__OSX__ -I../include -I../ext/tinyformat -I../ext/enoki/include -I../ext/tbb/include -I../ext/pybind11/include -I/Users/leroyv/miniconda3/envs/mitsuba2/include/python3.7m -flto=thin  -stdlib=libc++ -D_LIBCPP_VERSION -std=gnu++17 -fcolor-diagnostics -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -fPIC   -fno-stack-protector -fomit-frame-pointer -fno-math-errno -ffp-contract=fast -march=native -Wall -Wextra -Wno-unused-local-typedefs -fvisibility=hidden -Wdouble-promotion -MD -MT src/emitters/CMakeFiles/area-obj.dir/area.cpp.o -MF src/emitters/CMakeFiles/area-obj.dir/area.cpp.o.d -o src/emitters/CMakeFiles/area-obj.dir/area.cpp.o -c ../src/emitters/area.cpp
../src/emitters/area.cpp:99:61: error: no member named 'sample' in 'mitsuba::Texture<float, Spectrum<float, 4> >'
            std::tie(wavelength, spec_weight) = m_radiance->sample(
                                                ~~~~~~~~~~  ^
../src/emitters/area.cpp:168:19: note: in instantiation of member function 'mitsuba::AreaLight<float, mitsuba::Spectrum<float, 4> >::sample_ray' requested here
MTS_EXPORT_PLUGIN(AreaLight, "Point emitter")
                  ^
1 error generated.
[298/377] Linking CXX shared library src/libcore/python/core_scalar_spectral_ext.cpython-37m-darwin.so
```

--- a/src/emitters/area.cpp
+++ b/src/emitters/area.cpp
@@ -96,7 +96,7 @@ public:
         Spectrum spec_weight;
 
         if constexpr (is_spectral_v<Spectrum>) {
-            std::tie(wavelength, spec_weight) = m_radiance->sample(
+            std::tie(wavelength, spec_weight) = m_radiance->sample_spectrum(
                 si, math::sample_shifted<Wavelength>(wavelength_sample), active);
         } else {
             wavelength = zero<Wavelength>();
@@ -165,5 +165,5 @@ private:
 };
 
 MTS_IMPLEMENT_CLASS_VARIANT(AreaLight, Emitter)
-MTS_EXPORT_PLUGIN(AreaLight, "Point emitter")
+MTS_EXPORT_PLUGIN(AreaLight, "Area emitter")
 NAMESPACE_END(mitsuba)

